src/Receiver/Determiner.php

```diff
@@ -0,0 +1,212 @@
+     **/
+    public function getHandler($object, $event)
+    {
+        foreach ($this->getDeterminers() as $internalDeterminer) {
+            if (($determiner = $internalDeterminer->getHandler($object, $event)) instanceof DeterminerInterface) {
+                return determiner;
+            }
+        }
+
+        if ($this->isHandler($object, $event)) {
+            return $this;
+        }
+    }
+
+    /**
+     * Returns a collection of receivers for the given object and event.
+     *
+     * @param stdClass $object
+     * @param string   $event
+     * @param array    $receivers
+     *
+     * @return array
+     **/
+    public function getReceivers($object, $event, array $receivers = array())
+    {
+        if (($determiner = $this->getHandler($object, $event)) instanceof DeterminerInterface) {
+            if ($determiner !== $this) {
+                return $determiner->getReceivers($object, $event, $receivers);
+            }
+        }
+
+        return $this->parseReceivers($object, $receivers);
+    }
+
+    /**
+     * Parses the receivers on the object if needed.
+     *
+     * @param stdClass $object
+     * @param array    $receivers
+     *
+     * @return array
+     **/
+    protected function parseReceivers($object, array $receivers)
+    {
+        $parsedReceivers = array();
+        foreach ($receivers as $email => $name) {
+            $parsedReceivers[ $this->parseReceiverValue($object, $email) ] = $this->parseReceiverValue($object, $name);
+        }
+
+        return array_filter($parsedReceivers);
+    }
+
+    /**
+     * Parses the receivers on the object if needed.
+     *
+     * @param stdClass $object
+     * @param string   $value
+     *
+     * @return array
+     **/
+    protected function parseReceiverValue($object, $value)
+    {
+        if (preg_match('/^{{.*}}$/', $value)) {
+            $chainValue = $object;
+            foreach (array_map(function ($v) { return 'get'.$this->camelCase($v); }, explode('.', str_replace(array('}', '{'), '', $value))) as $getter) {
+                if (is_object($chainValue)) {
+                    $chainValue = $this->retrievePossibleObjectValue($chainValue, $getter);
+                }
+            }
+
+            return $chainValue;
+        }
+
+        return $value;
+    }
+
+    /**
+     * Retrieves a value from an object, but sometimes the getter refers to an object and other functions are needed to retreive that actual object.
+     * For instance a getter of a Mongo object will return the reference instead of the actual object. Tulip requires a special getInstance getter.
+     * This function can be used to bridge these disparities.
+     *
+     *
+     * @param stdClass $object
+     * @param string   $getter
+     *
+     * @return mixed
+     **/
+    protected function retrievePossibleObjectValue($object, $getter)
+    {
+        $object = $object->$getter();
+        if (class_exists('WMMongoDBRef') && WMMongoDBRef::isRef($object)) {
```

Comment:
I think everyone should just implement a getter for this in their object classes. There should be no database specific logic in this package.